heft-plugins/heft-lint-plugin/src/LintPlugin.ts

```diff
@@ -52,29 +54,43 @@ export default class LintPlugin implements IHeftTaskPlugin<ILintPluginOptions> {
   private _tslintToolPath: string | undefined;
   private _tslintConfigFilePath: string | undefined;
 
+  private _checkFix(taskSession: IHeftTaskSession, pluginOptions?: ILintPluginOptions): boolean {
+    let fix: boolean =
+      pluginOptions?.alwaysFix || taskSession.parameters.getFlagParameter(FIX_PARAMETER_NAME).value;
+    if (fix && taskSession.parameters.production) {
+      // Write this as a standard output message since we don't want to throw errors when running in
+      // production mode and "alwaysFix" is specified in the plugin options
+      taskSession.logger.terminal.writeLine(
+        'Fix mode has been disabled since Heft is running in production mode'
+      );
+      fix = false;
+    }
+    return fix;
+  }
+
+  private _getSarifLogPath(
+    heftConfiguration: HeftConfiguration,
+    pluginOptions?: ILintPluginOptions
+  ): string | undefined {
+    const relativeSarifLogPath: string | undefined = pluginOptions?.sarifLogPath;
+    const sarifLogPath: string | undefined =
+      relativeSarifLogPath && path.resolve(heftConfiguration.buildFolderPath, relativeSarifLogPath);
+    return sarifLogPath;
+  }
```

Comment:
Minor note: since neither of these functions reference `this` it is more efficient to have them be defined as free functions in the module scope.